Remove confusing torch::jit::RegisterOperators for custom ops

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#28229 Remove confusing torch::jit::RegisterOperators for custom ops**

We have `torch::RegisterOperators` for custom ops. `torch::jit::RegisterOperators` had a dual state of being able to register custom ops if called one way and being able to register pure JIT ops if called another way.
This is confusing because you end up in different operator libraries depending on which API exactly you're using.

This PR removes the ability for torch::jit::RegisterOperators to register custom ops and forces people to use the new torch::RegisterOperators.

This was already deprecated before but we now remove it.

Differential Revision: [D17981895](https://our.internmc.facebook.com/intern/diff/D17981895/)

smessmer committed:
Remove confusing torch::jit::RegisterOperators for custom ops

We have `torch::RegisterOperators` for custom ops. `torch::jit::RegisterOperators` had a dual state of being able to register custom ops if called one way and being able to register pure JIT ops if called another way.
This is confusing because you end up in different operator libraries depending on which API exactly you're using.

This PR removes the ability for torch::jit::RegisterOperators to register custom ops and forces people to use the new torch::RegisterOperators.

This was already deprecated before but we now remove it.

Differential Revision: [D17981895](https://our.internmc.facebook.com/intern/diff/D17981895/)

[ghstack-poisoned]

bddppq reviewed (commented):
There are a few places still using it:
https://github.com/pytorch/pytorch/blob/319ef3b/test/onnx/test_custom_ops.py#L22-L23
https://github.com/pytorch/pytorch/blob/319ef3b/test/test_cpp_extensions.py#L580
https://github.com/pytorch/glow/blob/1656b9a/torch_glow/src/PyTorchCommon.cpp#L54
https://github.com/pytorch/glow/blob/1656b9a/torch_glow/src/PyTorchCommon.cpp#L156
https://github.com/pytorch/glow/blob/1656b9a/torch_glow/src/PyTorchFileLoader.cpp#L106

smessmer committed:
Update on "Remove confusing torch::jit::RegisterOperators for custom ops"


We have `torch::RegisterOperators` for custom ops. `torch::jit::RegisterOperators` had a dual state of being able to register custom ops if called one way and being able to register pure JIT ops if called another way.
This is confusing because you end up in different operator libraries depending on which API exactly you're using.

This PR removes the ability for torch::jit::RegisterOperators to register custom ops and forces people to use the new torch::RegisterOperators.

This was already deprecated before but we now remove it.

Differential Revision: [D17981895](https://our.internmc.facebook.com/intern/diff/D17981895/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove confusing torch::jit::RegisterOperators for custom ops"


We have `torch::RegisterOperators` for custom ops. `torch::jit::RegisterOperators` had a dual state of being able to register custom ops if called one way and being able to register pure JIT ops if called another way.
This is confusing because you end up in different operator libraries depending on which API exactly you're using.

This PR removes the ability for torch::jit::RegisterOperators to register custom ops and forces people to use the new torch::RegisterOperators.

This was already deprecated before but we now remove it.

Differential Revision: [D17981895](https://our.internmc.facebook.com/intern/diff/D17981895/)

[ghstack-poisoned]
